docs/source/add_new_model.md

```diff
@@ -0,0 +1,573 @@
+# How to add a model to 🤗Transformers
+
+Adding a new model is often difficult and requires an in-depth knowledge of the 
+🤗Transformers repository and ideally also of the model's original repository.
+At Hugging Face, we are trying to empower the community more and more to add models 
+independently. Thus, for some new models that the community wants to be added 
+to 🤗Transformers, we create a customized *call-for-model-addition* that explains 
+step-by-step how to add the requested model. 
+With this *call-for-model-addition*, we want to teach a motivated and experienced 
+contributor of the community how to port a model to Transformers.
+
+If this sounds like something you would be interested in, feel free to check out 
+the currently open "calls-for-model-addition" [here](https://github.com/huggingface/transformers/tree/master/templates/adding_a_new_model/open_model_proposals/README.md) and to contact us.
+
+If selected, you will then work closely with one member of the Hugging Face team to
+integrate the model into 🤗Transformers. By doing so, you will both gain a 
+theoretical and deep practical understanding of the proposed model. But more importantly, 
+you will have made a major open-source contribution to 🤗Transformers. Along the way,
+you will:
+	
+- get insights into open-source best practices, 
+- understand the design principles of one of the most popular NLP libraries,
+- learn how to do efficiently test large NLP models,
+- learn how to integrate python utilities like `black`, `isort`, `make fix-copies` into a library 
+  to always ensure clean and readable code.
+
+We are obviously also more than happy if you want to add a model which can not be found in 
+the "calls-for-model-addition" folder. The following sections explain in detail how to add a new model. It might also be very helpful 
+to check-out already added models to see if those resemble the model you would like 
+to add [here](https://github.com/huggingface/transformers/pulls?q=is%3Apr+label%3A%22PR+for+Model+Addition%22+is%3Aclosed).
+
+To start with let's try to get a general overview of the Transformers library.
+
+## General overview of 🤗Transformers
+
+TODO
+
+- Mention some general design principles
+- Mention the single file policy
+- Mention the copy over abstraction policy
+- ...
+
+### Overview of models
+
+TODO
+
+- Add graphic
+
+### Overview of tokenizers
+
+TODO
+
+- Add graphic
+
+## Adding brand_new_bert to 🤗Transformers
+
+For exemplary purposes, we will call the model that we will add to 🤗Transformers `"brand_new_bert"`.
+  
+The following list is a summary of everything that has to be done to add a model and can be used 
+by you as a To-Do List:
+
+- [ ] Understood theoretical aspects
+- [ ] Prepared environment
+- [ ] Set up debugging environment of original repository
+- [ ] Created script that successfully runs forward pass using original repository and checkpoint
+- [ ] Successfully converted original checkpoint to Transformers checkpoint
```

Comment:
I would add - [] Successfully added Model in Transformers before this conversion script.